torchmdnet/models/tensornet.py

```diff
@@ -163,6 +169,17 @@ def __init__(
         self.activation = activation
         self.cutoff_lower = cutoff_lower
         self.cutoff_upper = cutoff_upper
+        self.additional_labels = additional_labels
+        # initialize additional methods as None if not provided, also used by module.py 
+        self.additional_methods = None
+        
+        if additional_labels is not None:
+            self.additional_methods = {}
+            for method_name, method_args in additional_labels.items():
+                # the key of the additional_methods is the label of the method (total_charge, partial_charges, etc.)
+                # this will be useful for static shapes processing if needed
+                self.additional_methods[method_args['label']] = {'name': method_name, 'method':self.initialize_additional_method(method_name, method_args)}
```

Comment:
I do not really have a better alternative for you, but the names "additional_method" and "initialize_additional_method" should be a little less generic I think.
Maybe "label_handlers" or "additional_label_method"?

Reply:
Mmmm not sure, repeating additional_labels too many times is redundant. `additional_methods` handles only additional_labels, the relationship seems quite straightforward and I would say it is quite generic

Reply:
The fact that "additional_methods" handles only additional labels is the problem. The name carries too little meaning of what the variable does. It is akin to calling a member variable "additional_variable".

Reply:
You could also get rid of the variable altogether and store everything in additional_labels, or move it to a separate class.
The way you are using it now reminds me of callbacks, you could all it "label_callbacks"

Reply:
okok let's move to label_callbacks